lib/chef/resource/windows_task.rb

```diff
@@ -223,13 +222,13 @@ def validate_start_time(start_time, frequency)
 
       def validate_user_and_password(user, password)
         if password_required?(user) && password.nil?
-          raise ArgumentError, %q{Cannot specify a user other than the system users without specifying a password!. Valid passwordless users: 'NT AUTHORITY\SYSTEM', 'SYSTEM', 'NT AUTHORITY\LOCALSERVICE', 'NT AUTHORITY\NETWORKSERVICE', 'BUILTIN\USERS', 'USERS'}
+          raise ArgumentError, "Cannot specify a user other than the system users without specifying a password!. Valid passwordless users: #{Chef::ReservedNames::Win32::Security::SID::SYSTEM_USER.join(', ')}"
```

Comment:
In the message if we use ```"'#{array.join("', '")```}'", as an interpolation; we may retain the single quotes. Please try this:

```
'#{Chef::ReservedNames::Win32::Security::SID::SYSTEM_USER.join("', '")}'
```
and then we can retain the old format.

It means, we would not need to alter single quotes in [spec#L1574](https://github.com/chef/chef/blob/master/spec/functional/resource/windows_task_spec.rb#L1574)